src/olympia/amo/sitemap.py

```diff
@@ -69,6 +68,51 @@ def location(self, item):
         return reverse(f'addons.{item.urlname}', args=[item.slug])
 
 
+class AddonVersionSitemap(Sitemap):
+    priority = 0.5
+    changefreq = 'weekly'
+    # i18n = True  # TODO: support all localized urls
+    item_tuple = namedtuple('VersionItem', ['last_updated', 'slug', 'page'])
+
+    def items(self):
+        addons = list(
+            Addon.objects.public()
+            .order_by('-last_updated')
+            .annotate(
+                versions_count=Count(
+                    'versions',
+                    filter=Q(
+                        versions__deleted=False,
+                        versions__files__status__in=amo.REVIEWED_STATUSES,
+                    ),
+                )
+            )
+            .values_list(
+                'last_updated',
+                'slug',
+                'versions_count',
+                named=True,
+            )
+        )
+        items = []
+        page_size = settings.REST_FRAMEWORK['PAGE_SIZE']
+        for addon in addons:
+            pages_needed = math.ceil(addon.versions_count / page_size)
```

Comment:
I'm sorry this wasn't caught sooner, but the All Versions page on frontend does not support pagination, so I don't think we need to (or should) worry about multiple pages for this `Sitemap`.

Reply:
d'oh! I should have remembered that mozilla/addons-frontend#7542 was wontfix'd.  I'm going to close this PR and start again because it doesn't need any of this complexity now.